### DISCUSSION TITLE
Config File Syntax for U/V Winds

### DISCUSSION THREAD

**Discussion by craigrj** [dtcenter/METplus#979] [Configuration] [answered]
I am verifying ensemble probabilities for winds > 15, 25, 35 using grid stat.  The analysis file (ECMWF) has winds in components (10U,10V).  The question is how do we set the observation field of the GridStat configuration file.   The fcst field is:
```
fcst = {
    field = [ 
              { name = "PROB"; level = "Z10"; 
                 prob = { name = "GUST"; thresh_lo = 15.; } GRIB2_pdt=5; }
              { name = "PROB"; level = "Z10";   
                 prob = { name = "GUST"; thresh_lo = 25.; } GRIB2_pdt=5; },
              { name = "PROB"; level = "Z10";   
                 prob = { name = "GUST"; thresh_lo = 35.; } GRIB2_pdt=5; },
              { name = "PROB"; level = "Z10";   
                 prob = { name = "GUST"; thresh_lo = 50.; } GRIB2_pdt=5; }
            ];
            cat_thresh=[==0.01];
 }
```
How would I code the observation field to match the fcst field where the ECMWF analysis data is in U/V components.  The component names (GRIB 1) is 10U for U winds and 10V for V winds.  I did not see a good example of this in the Users Manual.

Thanks
Bob

**Comment by jprestop** [comment]
Hi Bob.  I think @JohnHalleyGotway is the best person to answer your question. Please allow a few business days for a response.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi Bob, thanks for making the move over to GitHub Discussions!

I see you have a question about verifying the probability of wind speed exceeding some threshold versus the U and V wind components (named 10U and 10V), as defined in an ECMWF GRIB file. I retrieved some sample data from the ECMWF Copernicus site to do some testing with 10U and 10V data. And I found that unfortunately the existing logic in MET's library code does not support what you're trying to do.

Here's the relevant line in the MET library code:
https://github.com/dtcenter/MET/blob/main_v10.0/met/src/libcode/vx_data2d_grib/data2d_grib.cc#L565

If the requested variable does not exist in the input file, this line checks to see if the data being requested in wind direction or wind speed. But that check is VERY hard-coded, looking only for GRIB codes 31 for wind direction or 32 for wind speed. And if so, it searches the input GRIB files for the U and V components but only by name UGRD and VGRD.

So if you were to request WIND (for wind speed) to be retrieved from that ECMWF GRIB file, it'd look for variables named UGRD and VGRD which are obviously not present, and it'll error out. And there is no existing mechanism by which wind speed could automatically be derived on the fly from 10U and 10V. However one option would be using a python script to derive wind speed from 10U and 10V and serving it up in memory.
